app/messaging/update-agreement.js

```diff
@@ -17,5 +19,10 @@ module.exports = (cache) => {
       default:
         log(`Message from unknown message source ${source}. Purposefully doing nothing.`)
     }
+
+    await updateAgreementReceiver.completeMessage(msg)
+  } catch (err) {
```

Comment:
I've commented the same on the `ffc-sfi-calculator` PR about moving the error handling for messages that are not processed into the function being run (`agreementChanges`) to ensure the message is abandoned.